BennyTordrupVisma/Week1/Ass2/index.html

```diff
@@ -0,0 +1,124 @@
+<html>
+<head>
+	<title>Hello world with react</title>
+	<style type="text/css">
+	body {
+		width: 1000px;
+		margin: auto;
+	}
+	.error-message {
+		color: red;
+	}
+	</style>
+
+	
+	<script src="http://fb.me/react-0.14.7.min.js"></script>
+	<script src="http://fb.me/react-dom-0.14.7.min.js"></script>
+	<!-- <script src="http://cdnjs.cloudflare.com/ajax/libs/babel-standalone/6.4.4/babel.min.js"></script> -->
+	<script src="http://fb.me/JSXTransformer-0.12.2.js"></script>
+</head>
+<body>
+	<!-- 1. Create an inputfield that doesn't accept any numbers -->
+	<!-- 2. Find a way to submit and print the message -->
+	<!-- 3. Validation errors in inputfields should produce an error-message rather than a hello morld message -->
+	<!-- 4. Any error-message should be printed in red color -->
+	<!-- 5. Provide a function that returns all messages ever printed -->
+	<!-- 6. This must be solved by using only native Javascript and react -->
+	<!-- 7. All markup must be produced with react -->
+	<!-- 8. All messages must be printed in the order they where submitted -->
+
+	<div id="base"/>
+	
+	<script type="text/jsx">
+		var InputField = React.createClass({
+			render: function() {
+				return <div>
+						<input type="text" ref="input" />
+						<button onClick={this.handleValidate}>Commit</button>
+						<button onClick={this.showAll}>Show all</button>
+					</div>;
+			},
+			
+			handleValidate: function() {
+				var input = this.refs.input.getDOMNode().value;
+				this.props.validate(input);
+				this.refs.input.getDOMNode().value = '';
+			},
+			
+			showAll: function() {
+				this.props.displayAll();
+			}
+		});
+		
+		var Message = React.createClass({
+			render: function() {
+				return <div>{this.props.message}</div>
+			}
+		})
+		
+		var ErrorMessage = React.createClass({
+			render: function() {
+				return <div className="error-message">{this.props.errorMessage}</div>
+			}
+		})
+		
+		var MessageList = React.createClass({
+			render: function() {
+				return <div>
+							<h1>All messages</h1>
+							<ol>
+								{this.props.allMessages.map(function(message) {
+									return <Message message={message} />;
+								})}
+							</ol>
+						</div>
+			}
+		})
+		
+		var App = React.createClass({
+			getInitialState: function() {
```

Comment:
While being a nice place for setting defaults, `getInitialState` should be avoided because we then might create multiple sources of truth as the application grows.
In this case you could put some of them directly in the `render´ and you could handle others by exploding how the js type system works. (`errorMessage` don't have to exist if it is empty)